### PR TITLE
[WIP] Refactor and fix docstring check

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR

- Extracts docstring tools to helper module
- Refactors the docstring check
- Moves `autonomy analyse abci docstrings` to `autonomy analyse docstrings`
- Removes `--check` flag and makes the check default flow of execution
- Adds `--update` flag to specify update docstrings if necessary

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [x] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/analyse/abci/docstrings.py [hunk 1]
```diff
@@ -78,41 +76,26 @@ def docstring_abci_app(abci_app: Any) -> str:  # pylint: disable-msg=too-many-lo
     )
 
 
-def update_docstrings(
-    module_path: Path, docstring: str, abci_app_name: str, check: bool = False
-) -> bool:
```
Comment (angrybayblade) [inline autonomy/analyse/abci/docstrings.py:83]: This function was doing so many things at a time, so I refactored into multiple functions which serves one purpose

File: autonomy/cli/analyse.py [hunk 2]
```diff
@@ -124,37 +124,40 @@ def abci_group() -> None:
...
-        else:
-            click.echo("No update needed.")
-    except Exception as e:  # pylint: disable=broad-except
-        raise click.ClickException(str(e)) from e
+    if packages_dir.name != PACKAGES:
+        raise click.ClickException(
+            f"packages directory {packages_dir} is not named '{PACKAGES}'"
+        )
+
+    with sys_path_patch(packages_dir.parent):
```
Comment (angrybayblade) [inline autonomy/cli/analyse.py:142]: Same as fsm spec check, we append the packages directory to `sys.paths` temporarily

File: packages/valory/skills/termination_abci/rounds.py [hunk 3]
```diff
@@ -108,8 +108,22 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
         return None
 
 
-class PostTerminationTxAbciApp(AbciApp):
-    """The abci app running after the multisig transaction has been settled."""
+class PostTerminationTxAbciApp(AbciApp[Event]):
+    """PostTerminationTxAbciApp
```
Comment (angrybayblade) [inline packages/valory/skills/termination_abci/rounds.py:112]: Will this cause any problems @Adamantios ?

**Comment by angrybayblade** [comment]
Windows failure is caused by dependency installation failure

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**